### DISCUSSION TITLE
[NEW] AppFlowy's 6-month Technical Roadmap

### DISCUSSION THREAD

**Discussion by annieappflowy** [AppFlowy-IO/AppFlowy#1715] [General]
Hi all, I'm excited to share our product roadmap for Q1-Q2 2023, including three major updates:
- We will be able to insert databases into a page
- AppFlowy will support mobile platforms
- You will find more plugins for the community by the community

As always, we're open to your feedback and very much looking forward to working with you all!

![image](https://user-images.githubusercontent.com/12026239/212696613-6ae32976-995c-4f76-a77c-bf415cadbb56.png)

**Comment by heyakyra** [comment]
Are certain features going to be made closed-source proprietary, like 'grid version history' as shown in the [previous roadmap](https://github.com/AppFlowy-IO/AppFlowy/discussions/565#discussioncomment-4856800)? I hope AppFlowy can stay open source instead of open core
> [![roadmap](https://web.archive.org/web/20230322195128if_/https://user-images.githubusercontent.com/12026239/180709306-e1a42453-30d9-4582-a275-36f1a85da8c2.png)](https://github.com/AppFlowy-IO/AppFlowy/discussions/565#discussioncomment-4856800)

**Reply by heyakyra** [reply]
For everyone who followed this, there was an official reply which implies that open source as done by companies like Red Hat, Kaleidos, New Vector, etc. is not sustainable:

> There are certain features that would incur extra cost (potentially huge) to us, so they are likely not "free" forever. Otherwise, it would be excruciating for us to make the project sustainable in the long term. 

Of course that can make sense for features which themselves incur a subscription cost (for example, sending SMS messages through some third party integration) but accessing local data such as version history is critical and removing that feels like an anti-feature similar to Slack's limitation of conversation history.

https://discord.com/channels/903549834160635914/903553722804748309/1096621366205227018

**Reply by annieappflowy** [reply]
I'm not sure about your logic to arrive such implication and I don't agree: "there was an official reply which implies that open source as done by companies like Red Hat, Kaleidos, New Vector, etc. is not sustainable."

I assume you have good intention, but we don't see your feature request for the version history or any code contributions to our repos. Can you share your GitHub issue submitted to our repo explaining what use cases you have with AppFlowy, what kind of features and needs you want in the version history, and how you would benefit from such features. In addition, we'd like to know what kind of users would also benefit from your feature request.

Lastly, given that your org https://editshare.com/ is a video content management IT firm with IT capabilities, you should consider building such features yourself.

**Reply by annieappflowy** [reply]
<img width="1547" alt="image" src="https://github.com/AppFlowy-IO/AppFlowy/assets/12026239/0c4920e6-3e0d-49cd-9197-4379a81ef71d">

Again, as I replied to you on April 15th, if you'd like something to be built by us, you can submit a feature request or become a contributor yourself.

**Reply by heyakyra** [reply]
Thanks for the quick response—and I'm sorry if I misunderstood your message. I interpreted your reply as confirming that the grid version history would be proprietary. If I am mistaken, and you plan to stay fully open source, I would definitely get behind the project 100%, ahead of others like Focalboard, Tagspaces, Baserow, Grist, etc! I am sure many others agree.

I have only been asking about a feature that was already marked as "(closed source)" on the roadmap you shared last year, so I assume any new ticket would be redundant to the tasks you are already tracking. Since I did not get a reply for a month on the previous discussion, I tried again here. After another month went by with no reply again, I finally tried discord and I do appreciate that I was able to get some clarity (I thought). I'd been very excited about Appflowy since it hit Hackernews, but I'm just personally much less interested in open core projects, and since others seemed interested in the question, I wanted to link the two together so others have the context too.

**Reply by annieappflowy** [reply]
As you can check from my remarks on Jan 16, the version history didn't get built at all and is not included in the latest roadmap posted in this thread. You can consider it not planned. And you should see our open-source codebase as a single source of truth and always refer back to it when making claims.

We rely on feature requests and bug reports by the community to guide our direction. The version history has many variations from high-level aggregated actions to very granular transactional operations.  Some variations can be beneficial to application users while others can be commercialized by proprietary companies who use our codebase to sell commercial solutions without contributing back. Since we don't receive your feature request, we don't know what problems it would solve, thus good solutions to your needs. Again, I believe you have good intentions. If you can provide us with more context, we will consider prioritizing it.

We'd appreciate more specific use cases, customer needs, and meaningful contributions.

**Reply by suntorytimed** [reply]
@annieappflowy looking at https://github.com/AppFlowy-IO/AppFlowy-Collab the repository is currently not licensed under an open source license. Will this happen in the future and was just not done at the current state of development or will this incur some kind of non-commercial use licensing?

**Reply by annieappflowy** [reply]
Thanks @suntorytimed for bringing it up. We forgot to add the license. Now it's under AGPL-3.0
